deptry/cli.py

```diff
@@ -37,6 +41,46 @@ def convert(
 COMMA_SEPARATED_TUPLE = CommaSeparatedTupleParamType()
 
 
+class CommaSeparatedMappingParamType(click.ParamType):
+    """
+    This class is used to uniformly handle configuration parameters that can be either passed as a comma-separated pair
+    string, or as a Mapping of strings to tuples of strings. Items in a pair string are separated by an equal sign,
+    where multiple values are separated by a pipe: key1=value1,key2=value2|value3.
+    For example, the value for a parameter can be a comma-separated pair string when passed as a command line argument,
+    or as a mapping of string to tuples of strings when passed through pyproject.toml.
+    """
+
+    name = "mapping"
+
+    def convert(
+        self,
+        value: str | Mapping[str, tuple[str, ...]],
```

Comment:
Should accept `Mapping[str, list[str]]` too, as toml produces lists, not tuples.